From 4036019821be1f94bed724e2ebdb2a3b6ff41670 Mon Sep 17 00:00:00 2001
From: Adrien Destugues <pulkomandy@pulkomandy.tk>
Date: Thu, 10 May 2018 10:21:34 +0200
Subject: Reorder some includes to avoid conflicts between zlib and openssl

zlib #defines free_func, but ssl headers use free_func in a parameter
name in some places. This ends up confusing gcc2. Make sure the ssl
headers are included before the define is set, so we can build libgit
with gcc2.


diff --git a/src/fetch.c b/src/fetch.c
index 0b22b36..67fdfbf 100644
--- a/src/fetch.c
+++ b/src/fetch.c
@@ -5,6 +5,7 @@
  * a Linking Exception. For full terms see the included COPYING file.
  */
 
+#include "netops.h"
 #include "fetch.h"
 
 #include "git2/oid.h"
@@ -15,7 +16,6 @@
 #include "remote.h"
 #include "refspec.h"
 #include "pack.h"
-#include "netops.h"
 #include "repository.h"
 #include "refs.h"
 
diff --git a/src/pack-objects.h b/src/pack-objects.h
index c9cd577..419e8a5 100644
--- a/src/pack-objects.h
+++ b/src/pack-objects.h
@@ -8,12 +8,13 @@
 #ifndef INCLUDE_pack_objects_h__
 #define INCLUDE_pack_objects_h__
 
+#include "netops.h"
+
 #include "common.h"
 
 #include "buffer.h"
 #include "hash.h"
 #include "oidmap.h"
-#include "netops.h"
 #include "zstream.h"
 #include "pool.h"
 #include "indexer.h"
diff --git a/src/push.c b/src/push.c
index 85b683e..9eab56e 100644
--- a/src/push.c
+++ b/src/push.c
@@ -5,12 +5,13 @@
  * a Linking Exception. For full terms see the included COPYING file.
  */
 
+#include "pack-objects.h"
+
 #include "push.h"
 
 #include "git2.h"
 
 #include "pack.h"
-#include "pack-objects.h"
 #include "remote.h"
 #include "vector.h"
 #include "tree.h"
-- 
2.16.2

